resolver/addressable_resolver.go

```diff
@@ -148,6 +153,17 @@ func (r *URIResolver) URIFromObjectReference(ctx context.Context, ref *corev1.Ob
 		return nil, apierrs.NewBadRequest("ref is nil")
 	}
 
+	// try custom resolvers first
+	for _, resolver := range r.resolvers {
+		url, err := resolver(ctx, ref)
+		if err != nil {
+			return nil, err
+		}
+		if url != nil {
```

Comment:
here is where the returned bool would come in handy and you don't have to guess if a nil url with a nil error means skip. Which is not documented behaviour and not expected.